### ISSUE TITLE
Cannot connect to TightVNC Server When Unauthenticated 

### ISSUE THREAD

**Issue by TheCuriousGuy** [mRemoteNG/mRemoteNG#274]
|||
|--:|---|
|Operating system | Windows 8 x64 |
|mRemoteNG version| Stable 1.74 Final  |

Connot connect to TightVNC using VNC protocol when Vnc Server is running in unauthenticated mode, i.e. no password is supplied to MRemoteNG.

Although reported as fixed in https://mremoteng.atlassian.net/browse/MR-198
the issue persists when not supplying password, if used in authenticated mode the connection works as expected.

when connecting for the first time (no pwd supplied) following error is reported:

----------
ErrorMsg
11/28/2016 4:10:44 PM
Opening connection failed!
Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host.
----------

----------
Opening connection failed!
Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host.
---------

However once text is inserted to pwd, a connection atemopt is excuted and then pwd is rolled back,
error returns to original reported issue:

----------
ErrorMsg
11/28/2016 4:13:29 PM
Opening connection failed!
Parameter is not valid.
----------

**Comment by sparerd** [comment]
Thanks for the report, we will need to dig into this

**Comment by TheCuriousGuy** [comment]
Much appreciated

i know its not a common use case, but internally with many user ion lab
environments,

my team prefers to not use pwd and thus this problem came up.

Cheers

**Comment by kmscode** [comment]
From user @vargasjj via #396 

Whenever I tried to VNC to a TightVNC server that does not have a password set, I get:

> ErrorMsg Opening connection failed! Parameter is not valid.

This defect is being around on mRemoteNG for a while and I cannot find a resolution. I can reproduce it.

Here are the logs:

> 2017-02-09 09:34:06,649 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)
2017-02-09 09:34:20,399 [1] ERROR- Opening connection failed!
Parameter is not valid.
2017-02-09 09:34:20,419 [258] INFO - Connection Event Closed
2017-02-09 09:34:20,420 [258] INFO - Connection to x.x.x.x via VNC closed by user ******.
2017-02-09 09:34:20,571 [258] ERROR- Couldn't dispose control, probably form is already closed (Connection.Protocol.Base)
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at System.Windows.Forms.Control.Invoke(Delegate method)
   at mRemoteNG.Connection.Protocol.ProtocolBase.DisposeControl()
   at mRemoteNG.Connection.Protocol.ProtocolBase.CloseBG()
2017-02-09 09:34:20,574 [258] ERROR- Couldn't set InterfaceControl.Parent.Tag or Dispose Interface, probably form is already closed (Connection.Protocol.Base)
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at System.Windows.Forms.Control.Invoke(Delegate method)
   at mRemoteNG.Connection.Protocol.ProtocolBase.DisposeInterface()
   at mRemoteNG.Connection.Protocol.ProtocolBase.CloseBG()
2017-02-09 09:38:39,990 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.clipboardchangeEventHandler.Invoke()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)
2017-02-09 09:38:39,992 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.clipboardchangeEventHandler.Invoke()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)
2017-02-09 09:39:00,541 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.clipboardchangeEventHandler.Invoke()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)
2017-02-09 09:49:02,050 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.clipboardchangeEventHandler.Invoke()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)
2017-02-09 09:49:02,444 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.clipboardchangeEventHandler.Invoke()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)
2017-02-09 09:50:45,428 [1] ERROR- frmMain WndProc failed
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at VncSharp.VncClient.OnConnectionLost()
   at VncSharp.VncClient.WriteClientCutText(String text)
   at VncSharp.RemoteDesktop.FillServerClipboard()
   at mRemoteNG.Connection.Protocol.VNC.ProtocolVNC.VNCEvent_ClipboardChanged()
   at mRemoteNG.UI.Forms.frmMain.clipboardchangeEventHandler.Invoke()
   at mRemoteNG.UI.Forms.frmMain.WndProc(Message& m)

Since you guys took the project now to Github I was wondering if you have any fix soon, or a workaround.

I searched here on github and on your reddit page. I think the referred links are related but I am not sure.

**Comment by vargasjj** [comment]
Hi, guys.

Just checking, if you'd tackle this issue on this quarter, or in the near future?

Thank you.

- JJ

**Comment by deusxanime** [comment]
Trying to rez this... I run into this problem when trying to connect to my KVM virtual machines that I run in unRAID. I don't usually bother with VNC password especially on initial install/config.

When I try to connect I get "invalid parameter" error as others here and on previous support forums were getting.

**Comment by BasicHumanUnit** [comment]
I also have issues with this. Specifically with our Timeclocks that allow for a vnc connection, I cannot set a password on them. Can connect no problem with vncviewer. I have no issues connecting to other devices that have a password on them

**Comment by tylla** [comment]
We met the same problem with some older VM's (qemu) which are unable to set a password for console view.

**Comment by vargasjj** [comment]
@tylla @deusxanime 

> We met the same problem with some older VM's (qemu) which are unable to set a password for console view.
--


> Trying to rez this... I run into this problem when trying to connect to my KVM virtual machines that I run in unRAID. I don't usually bother with VNC password especially on initial install/config.

When I try to connect I get "invalid parameter" error as others here and on previous support forums were getting.


Exactly. That is what I am trying to achieve for the last 3y because I use KVM (actually is Unetlab) and the embedded VMs (QEMU machines, Win7, Tinycore) uses this TightVNC Server that has no authentication by default.

Hey @sparerd I know you guys are probably busy on some other priority bugs. But was wondering if you have any plans for this.

Thanks. I love mRemoteNG. You guys are the best.
Cheers

~ JJ

**Comment by codetuner** [comment]
Issue of getting error "Parameter is not valid." when connecting using VNC without password authentication still present in mRemoteNG 1.75.7012.16814.
Just so you know.

**Comment by vargasjj** [comment]
Hello everyone

Issue of getting "Parameter is not valid" still present on mRemoteNG 1.76.10.42480 Portable.
----------
ErrorMsg
10/17/2018 08:56:15
Opening connection failed!
Parameter is not valid.
----------

@sparerd @kmscode would you have any idea if there's any road map for this issue? or perhaps can you help us trying to raise some visibility on this issue if possible?
I love mRemoteNG and I know it is not a commercial project so no push. It's just that it's being around for 6 years so far because it has never been addressed as mentioned by the OP @TheCuriousGuy  (https://mremoteng.atlassian.net/browse/MR-198)

**Comment by bithost** [comment]
Have same problem when connecting to VNC without password. mRemoteNG version: 1.76.11.40527

**Comment by TopBanana123** [comment]
Same issue as everyone else, I have worked around the issue by making an "External tool" which points to TightVNC with the arguments to connect to the host, it works fine

![image](https://user-images.githubusercontent.com/43852644/52538489-dbc43c00-2d38-11e9-8236-241d189d00d8.png)

![image](https://user-images.githubusercontent.com/43852644/52538498-00b8af00-2d39-11e9-9088-26cce091e98f.png)

**Comment by NextBlaubeere** [comment]
still an issue in 2019. is there any plan to fix this?

**Comment by sparerd** [comment]
I think we have this solved on the most recent build of vncsharp (which mRemoteNG uses). I am going to schedule this for the next release (v1.77) for testing.

**Comment by NextBlaubeere** [comment]
I updated to the current dev-branch (1.77.0.41252) but it seems still there.
When attempting to open the vnc-connection there is always the error message: **_"Opening connection failed! Parameter is not valid."_**

Log file states this:
> 2019-05-31 09:24:19,192 [1] INFO  - mRemoteNG 1.77.0.41252 starting.
2019-05-31 09:24:19,203 [1] INFO  - Command Line: C:\Program Files (x86)\mRemoteNG\mRemoteNG.exe
2019-05-31 09:24:19,478 [1] INFO  - Microsoft Windows 10 Pro  64-bit
2019-05-31 09:24:19,478 [1] INFO  - Microsoft .NET CLR 4.0.30319.42000
2019-05-31 09:24:19,479 [1] INFO  - System Culture: de-DE/Deutsch (Deutschland)
2019-05-31 09:24:19,479 [1] INFO  - Checking FIPS Policy...
2019-05-31 09:24:19,480 [1] INFO  - Checking Lenovo AutoScroll Utility...
2019-05-31 09:24:19,521 [1] INFO  - Override Culture: en/English
2019-05-31 09:24:19,616 [1] INFO  - Loading External Apps from: C:\Users\Manuel\AppData\Roaming\mRemoteNG\extApps.xml
2019-05-31 09:24:19,618 [1] INFO  - Adding External App: WinSCP C:\Program Files (x86)\WinSCP\WinSCP.exe scp://%Username%:%Password%@%Hostname%/
2019-05-31 09:24:20,923 [1] WARN  - PuttySessions.Watcher.StartWatching() failed.
Nicht gefunden 
2019-05-31 09:24:20,923 [1] WARN  - XmingPortablePuttySessions.Watcher.StartWatching() failed: 'sessions' does not exist.
2019-05-31 09:24:23,284 [1] ERROR - Opening connection failed!
Parameter is not valid.
2019-05-31 09:24:23,306 [1] ERROR - Protocol Event Disconnected failed.
{0}
Unable to cast object of type 'VncSharp.RemoteDesktop' to type 'mRemoteNG.Connection.Protocol.ProtocolBase'.
   at mRemoteNG.Connection.ConnectionInitiator.Prot_Event_Disconnected(Object sender, String disconnectedMessage, Nullable'1 reasonCode) in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\Connection\ConnectionInitiator.cs:line 252
2019-05-31 09:24:23,307 [10] INFO  - Connection Event Closed
2019-05-31 09:24:23,308 [10] INFO  - Connection to 192.168.1.152 via VNC closed by user Manuel.
2019-05-31 09:24:23,314 [9] INFO  - Connection Event Closed
2019-05-31 09:24:23,314 [9] INFO  - Connection to 192.168.1.152 via VNC closed by user Manuel.
2019-05-31 09:24:25,177 [1] ERROR - Opening connection failed!
Parameter is not valid.
2019-05-31 09:24:25,183 [1] ERROR - Protocol Event Disconnected failed.
{0}
Unable to cast object of type 'VncSharp.RemoteDesktop' to type 'mRemoteNG.Connection.Protocol.ProtocolBase'.
   at mRemoteNG.Connection.ConnectionInitiator.Prot_Event_Disconnected(Object sender, String disconnectedMessage, Nullable'1 reasonCode) in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\Connection\ConnectionInitiator.cs:line 252
2019-05-31 09:24:25,184 [15] INFO  - Connection Event Closed
2019-05-31 09:24:25,185 [15] INFO  - Connection to 192.168.1.152 via VNC closed by user Manuel.
2019-05-31 09:24:25,189 [14] INFO  - Connection Event Closed
2019-05-31 09:24:25,190 [14] INFO  - Connection to 192.168.1.152 via VNC closed by user Manuel.

**Comment by XXtreem11** [comment]
This is becoming more and more needed with the changes in Xorg where the VNC sessions and displays are different between the login screen and the actual UI.  For my Ubuntu login screen, the VNC port is 5920 without a password.  When I'm actually logged in, my desktop VNC port is 5900 with a password.   It would be great if mRemoteNG could support this for ease of use.

**Comment by zaphod414** [comment]
Holy crap, this issue is at least 7 YEARS old!  It is documented at least as far back as March 2013.  That is just plain pathetic.  It's one thing to be broken.  It's another thing entirely to stay broken year after year after year.  So long mRemoteNG.

**Comment by farosch** [comment]
@zaphod414 Yes, thats really sad. And all because of jerks like you that use an open source product but refuse to participate into the community but expect that three people maintain a software in their free time

**Comment by damntourists** [comment]
bump

I'm also experiencing this issue when attempting to connect to a vnc server running on VMs in my Proxmox cluster. It works fine for other software, however, crashes on mRemoteNG and it seems to cause it to crash on the VMs as well, as I'm no longer able to VNC to those VMs after an attempt with mRemoteNG until the node is rebooted.

**Comment by ewiegers** [comment]
This is is still an issue.  Is this not easy to fix?  My use case, as with others, is on an internal test environment.

**Comment by omegasupreme3** [comment]
Just to let you all know.

I build the software solution myself and I found that this issue doesn't occur in a Debug build. In my situation it only occurs when running a Release mode build. I did some research as to what happens, but so far I did not find something conclusive.

**Comment by NiKiZe** [comment]
99% of all my VNC connections are without password (to localhost which is a ssh tunnel to qemu instances)
This being an issue open for so many years means that devs don't have much time to spend on the project, or is it something else going on?

**Comment by Kvarkas** [comment]
> This being an issue open for so many years means that devs don't have much time to spend on the project, or is it something else going on?

if that so easy to solve could you raise a PR will be happy to merge, overwise lets wait till somebody else will find a solution for such...

**Comment by Luk164** [comment]
Same issue, does anyone know of any workaround?

**Comment by fgclue** [comment]
11:25
28/05/2022

**Comment by Luk164** [comment]
> ```
>       11:25
> ```
> 
> 28/05/2022

??? What does that mean?

**Comment by mlacomb** [comment]
Still an issue.  1.76.20.24615 Windows

**Comment by techie2000** [comment]
I ran into this bug a few years ago that stopped me using it (passwordless vnc is 70-80% of my usecase). I might give @TopBanana123 https://github.com/mRemoteNG/mRemoteNG/issues/274#issuecomment-462164438 workaround a go, even if it is an unwelcome faff 😀

**Comment by YellerFeller** [comment]
I am encountering this issue on version 1.76.20.24615 (Windows). It works after setting a password, but I would prefer to not have to do this.

**Comment by fgclue** [comment]
> > ```
> >       11:25
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 28/05/2022
> 
> ??? What does that mean?

I have no idea, I posted that like a year ago and now I'm confused too.

**Comment by jan-oratowski** [comment]
**TL;DR;**
It works in the latest nightly, however, there is another bug in VncSharpCore that makes it impossible to establish any VNC connection - created PR for that in VncSharpCore repo.

![image](https://github.com/mRemoteNG/mRemoteNG/assets/54535207/23ba0ea7-1d63-41cc-8b71-da019f15a7ba)

**Longer story:**
I wanted to fix it, so I:
 - downloaded mRemoteNG repo, switched branch to v1.77.3-dev
 - downloaded VncSharpCore repo,
 - in mRemoteNG solution removed reference to VncSharpCore nuget and replaced it with VncSharpCore project
 - compiled in debug first, tested (there was an issue with one resource file missing in VncSharpCore, fixed in PR) - works
 - compiled in release, tested - works as well

**Comment by Kvarkas** [comment]
@jan-oratowski Thanks a lot for such, let me update dev branch with such.

**Comment by Gabtoof** [comment]
> Same issue as everyone else, I have worked around the issue by making an "External tool" which points to TightVNC with the arguments to connect to the host, it works fine

@TopBanana123 In case you weren't aware (base off your screenshot), just letting you know that rather than putting a specific hostname in the External Tool config, you can instead put %Hostname% so that the one External Tool entry will work for any connection.

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by robertpopa22** [comment]
**Fix available for testing**

**Commit:** `30217039` (fork `robertpopa22/mRemoteNG`, branch `main`)
**What changed:** Cannot connect to TightVNC Server When Unauthenticated 

A beta build (v1.81.0-beta.4) including this fix is available from the fork's Releases page.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
